src/week2.unique-string/papakang22.js

```diff
@@ -0,0 +1,34 @@
+
+//Completed in 568ms
+function findUniq(arr) {
+    // do magic
+    let answer;
+    for(let i = 1; i < arr.length -1 ; i++){
+      
+      const strSet = new Set(arr[i].toLowerCase().split("")); //소문자로 변경 후 배열만든다음 중복을 제거
+      const nextSet = new Set(arr[i+1].toLowerCase().split("")); //소문자로 변경 후 배열만든다음 중복을 제거
+      const prevSet = new Set(arr[i-1].toLowerCase().split("")); //소문자로 변경 후 배열만든다음 중복을 제거
+      
+      const uniqStrArr = [...strSet]; //중복을 제거된 배열 생성
+      const uniqNextStrArr = [...nextSet]; //중복을 제거된 배열 생성
+      const uniqPrevStrArr = [...prevSet]; //중복을 제거된 배열 생성
+      
+      const str = uniqStrArr.sort().join("").trim(); //정렬후 문자열로 변환 및 공백제거
+      const nextStr = uniqNextStrArr.sort().join("").trim(); //정렬후 문자열로 변환 및 공백제거
+      const prevStr = uniqPrevStrArr.sort().join("").trim(); //정렬후 문자열로 변환 및 공백제거
+      
+      if(str != nextStr && str != prevStr){ // 양옆과 다르면 현재 순번인 문자열이 유니크값
+        answer = arr[i];
+        break;
+      }else if(str != nextStr && str == prevStr){ // 다음 문자열만 다르면 다음 순번인 문자열이 유니크값
+        answer = arr[i+1];
+        break;
+      }else if(str == nextStr && str != prevStr){ // 이전 문자열만 다르다면 이전 순번인 문자열이 유니크값
+        answer = arr[i-1];
+        break;
+      }
+    }
```

Comment:
array/set 사이 변환이나 sort 같은 코스트가 있는 작업을 수행하고 있으니 메모이제이션을 써 볼 수 있을 것 같아요. `if break` 다음에도 `else`는 생략 가능해 보이구요.

```suggestion
    const temp = arr.map(item => {
      const strSet = new Set(item.toLowerCase().split("")); //소문자로 변경 후 배열만든다음 중복을 제거
      const uniqStrArr = [...strSet]; //중복을 제거된 배열 생성
      const str = uniqStrArr.sort().join("").trim(); //정렬후 문자열로 변환 및 공백제거
      return str;
    });
    for(let i = 1; i < arr.length -1 ; i++){
      const str = temp[i];
      const nextStr = temp[i + 1];
      const prevStr = temp[i - 1];
      
      if (str != nextStr && str != prevStr){ // 양옆과 다르면 현재 순번인 문자열이 유니크값
        answer = arr[i];
        break;
      }
      if (str != nextStr && str == prevStr){ // 다음 문자열만 다르면 다음 순번인 문자열이 유니크값
        answer = arr[i+1];
        break;
      }
      if (str == nextStr && str != prevStr){ // 이전 문자열만 다르다면 이전 순번인 문자열이 유니크값
        answer = arr[i-1];
        break;
      }
    }
```

Reply:
아 미리 배열을 만들어서 하는방법이 있었군요 왜 생각을 못했을까욬ㅋㅋ 감사합니다. 메모이제이션이란 단어도 처음 듣는단어네요!! 공부 더 열심히 해야겠어여